collect_app/src/test/java/org/odk/collect/android/utilities/ArrayUtilsTest.java

```diff
@@ -0,0 +1,41 @@
+package org.odk.collect.android.utilities;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class ArrayUtilsTest {
+    @Test
+    public void toPrimitive() throws Exception {
+        Long[] object = new Long[] {1L, 2L, 3L};
+        long[] primitive = new long[] {1, 2, 3};
+        assertArrayEquals("toPrimitiveTest", primitive, ArrayUtils.toPrimitive(object));
+
+        long[] emptyLongPrimitive = new long[0];
+        assertArrayEquals("toPrimitiveNullTest", emptyLongPrimitive, ArrayUtils.toPrimitive(null));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void nullToPrimitive() {
+        Long[] object = new Long[] {1L, null, 3L, 4L, 5L};
+        ArrayUtils.toPrimitive(object);
```

Comment:
Are all three of these calls to toPrimitive expected to result in NPEs? Do the second and third calls get made, or does the test end after the first?

Reply:
@dcbriccetti Thanks for pointing it out.
Only first call gets executed and test ends.
Second and third calls are not executed.

Reply:
There are 2 solutions:
1. Write a separate test for each case.
2. In a single test, use multiple try and catch blocks for each function call.